### PR TITLE
GEODE-9063: Autocomplete with mandatory params first

### PR DESCRIPTION
**Pull request by alb3rtobr**

### CODE REVIEW

File: geode-gfsh/src/integrationTest/java/org/apache/geode/management/internal/cli/GfshParserConverterTest.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public void testJsonKey() {
   public void testUnspecifiedValueToStringArray() {
     String command = "change loglevel --loglevel=finer --groups=group1,group2";
     ParseResult result = parser.parse(command);
-    String[] memberIdValue = (String[]) result.getArguments()[0];
```
Comment (mhansonp) [inline geode-gfsh/src/integrationTest/java/org/apache/geode/management/internal/cli/GfshParserConverterTest.java:106]: Why did this switch to 1?

Comment (alb3rtobr) [reply geode-gfsh/src/integrationTest/java/org/apache/geode/management/internal/cli/GfshParserConverterTest.java:106]: Due to the change in the order of parameters in `ChangeLogLevelCommand`, the memberId is returned in second position by the parser, so I had to modify the array index from 0 to 1.

**Review by jinmeiliao — APPROVED** [review]
LGTM

File: geode-gfsh/src/integrationTest/java/org/apache/geode/management/internal/cli/GfshParserAutoCompletionIntegrationTest.java [hunk 2]
```diff
@@ -447,4 +447,124 @@ public void testCompletionCreateGatewaySenderWithDash() {
     assertThat(candidate.getCandidates()).hasSize(1);
     assertThat(candidate.getFirstCandidate()).isEqualTo(buffer + "id");
   }
+
+  @Test
+  public void forChangeLogLevelWithSpaceCompletionOffersMandatoryOptionsInAlphabeticalOrder() {
```
Comment (albertogpz) [inline geode-gfsh/src/integrationTest/java/org/apache/geode/management/internal/cli/GfshParserAutoCompletionIntegrationTest.java:452]: I would rename this test case and the rest to align them with the ones already testing something similar, like for example, testCompletionCreateGatewaySenderWithSpace.

In this case, the name would be: testCompletionChangeLogLevelWithSpace.
If you want to add also the expected outcome in the name like you already have in your original proposal, better.

This comments applies to all new test cases.

**Review by albertogpz — CHANGES_REQUESTED** [review]

**Review by albertogpz — APPROVED** [review]
Nice changes. Thanks!

**Pull request merged**